great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -614,7 +614,9 @@ def list_datasources(self) -> List[dict]:
         datasource_name: str
         datasource_config: DatasourceConfig
         for datasource_name, datasource_config in self.config.datasources.items():
-            datasource_dict: dict = datasource_config.to_json_dict()
+            datasource_dict: dict = cast(
+                dict, datasourceConfigSchema.dump(datasource_config)
+            )
```

Comment:
Need to go through schema validation instead of `to_json_dict` to get rid of null ids

Reply:
why is a cast required here?

Reply:
Marshmallow returns `Any` for dump - this doesn't change runtime behavior but let's our IDE's know what to actual expect.

We could type annotate `dump` as `dict` but we don't want to modify vendored code.